### PR TITLE
Nomos chat app non interactive

### PR DESCRIPTION
**Pull request by bacv**
Add params to nomos cli app to send chat message and exit the application. This will be used alongside the dissemination app to produce constant traffic in the testnet.

For chat app to decode messages, they need to be in ChatMessage struct, encoded with bincode. Bincode has no external cli tool to encode independantly, so either we need to write that tool, that we would be able to use from scripts, or we use chatapp to do encoding to ChatMessage directly, because it's specific to it.

Dissemination app will still be used for big data blob uploads to the testnet.

### CODE REVIEW

File: nomos-cli/src/cmds/disseminate/mod.rs [hunk 1]
```diff
@@ -44,15 +44,15 @@ impl Disseminate {
...
-            (None, None) => return Err("Either data or file must be specified".into()),
+
+        let bytes: Box<[u8]> = if let Some(data) = &self.data {
+            data.clone().as_bytes().into()
+        } else {
+            let file_path = self.file.as_ref().unwrap();
+            let file_bytes = std::fs::read(file_path)?;
+            file_bytes.into_boxed_slice()
         };
+
```
Comment (zeegomo) [inline nomos-cli/src/cmds/disseminate/mod.rs:55]: nit: why this change?

**Review by zeegomo — APPROVED** [review]

File: nomos-cli/src/cmds/disseminate/mod.rs (hunk 1 above)
Comment (bacv) [reply nomos-cli/src/cmds/disseminate/mod.rs:55]: The previous solution, even being readable and concise, the panic message differed from the user friendly warning that Clap produced. The output when user provides the `--author` flag, but forgets the `--message` now is:
```
error: the following required arguments were not provided:
  --message <MESSAGE>

Usage: nomos-cli chat --network-config <NETWORK_CONFIG> --node <NODE> --message <MESSAGE> --author <AUTHOR>
```
If user doesn't provide the author nor message, the app goes into the interactive mode

**Review by youngjoon-lee — APPROVED** [review]
Looks good!

> Bincode has no external cli tool to encode independantly, so either we need to write that tool, that we would be able to use from scripts, or we use chatapp to do encoding to ChatMessage directly, because it's specific to it.

in my opinion, it's good enough to do encoding in the chatapp, as you implemented.
